src/sinks/splunk_hec/metrics.rs

```diff
@@ -0,0 +1,713 @@
+use crate::{
+    config::{DataType, GenerateConfig, SinkConfig, SinkContext, SinkDescription},
+    event::{Event, Metric, MetricValue},
+    internal_events::SplunkInvalidMetricReceived,
+    internal_events::{SplunkEventEncodeError, SplunkEventSent},
+    sinks::splunk_hec::conn,
+    sinks::util::{encode_namespace, http::HttpSink, BatchConfig, Compression, TowerRequestConfig},
+    sinks::{Healthcheck, VectorSink},
+    template::Template,
+    tls::TlsOptions,
+};
+use http::Request;
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use std::borrow::Cow;
+use std::collections::BTreeMap;
+use std::iter;
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct HecSinkMetricsConfig {
+    pub default_namespace: Option<String>,
+    pub token: String,
+    pub endpoint: String,
+    #[serde(default = "host_key")]
+    pub host_key: String,
+    pub index: Option<Template>,
+    pub sourcetype: Option<Template>,
+    pub source: Option<Template>,
+    #[serde(default)]
+    pub compression: Compression,
+    #[serde(default)]
+    pub batch: BatchConfig,
+    #[serde(default)]
+    pub request: TowerRequestConfig,
+    pub tls: Option<TlsOptions>,
+}
+
+#[derive(Serialize, Debug, PartialEq)]
+#[serde(untagged)]
+enum FieldValue<'a> {
+    Str(Cow<'a, str>),
+    Float(f64),
```

Comment:
I see this is currently only serialized. Will this ever need to be deserialized? If so, the variants will need to be in the other order.

Reply:
I think it's unlikely, but It seems sensible to make this change regardless